### ISSUE TITLE
What is right steps to re-generate edgeHub's certificate after change HostName on config.toml 

### ISSUE THREAD

**Issue by KevinKao809** [Azure/iotedge#5773]
## Expected Behavior
The edgeHub present certificate be updated with new HostName on CN.

## Current Behavior
edgeHub still present old certificate 

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. In config.toml, uncomment hostname, and change it to be "abc.com"
2. apply the change and restart iotedge.
3. when edgeHub ready, verify it present certificate by command: openssl s_client -connect abc.com:8883
4.

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```

<Paste here between the triple backticks>

```
</details>

### Device Information
* Host OS [Debian 9]: 
* Architecture [arm32]: 
* Container OS []: 

### Runtime Versions
* aziot-edged [1.2.4]: 
* Edge Agent [1.2]: 
* Edge Hub [1.2]: 
* Docker/Moby [3.0.5]: 

Note: when using Windows containers on Windows, run `docker -H npipe:////./pipe/iotedge_moby_engine version` instead

## Logs

<details>
<summary>aziot-edged logs</summary>

```

<Paste here between the triple backticks>

```
</details>

<details>
<summary>edge-agent logs</summary>

```

<Paste here between the triple backticks>

```
</details>

<details>
<summary>edge-hub logs</summary>

```

<Paste here between the triple backticks>

```
</details>

## Additional Information
Please provide any additional information that may be helpful in understanding the issue.

**Comment by pebneter** [comment]
Did you ever find a definitive answer?
I also struggled with this one several times. Just recently I had the issue and then tried several setting and in the end managed to fix it. But I'm still unsure what setting finally did the trick.

**Comment by jlian** [comment]
Hey we added some docs here https://docs.microsoft.com/en-us/azure/iot-edge/how-to-connect-downstream-iot-edge-device?view=iotedge-2020-11&tabs=azure-portal#verify-parent-configuration

Essentially, try `sudo docker rm -f edgeAgent`

The reason why this is needed is that containers need to be recreated for the environment variables (including the hostname) to be updated. Removing `edgeAgent` is the easiest way to do that. We know this is an unexpected behavior and we merged a fix https://github.com/Azure/iotedge/pull/6372 to prevent you from getting into this state.
